moonstone/filtering/basics_filtering.py

```diff
@@ -61,3 +63,72 @@ def filter(self) -> pd.DataFrame:
             return self._select_names()
         else:
             return self._exclude_names()
+
+
+class NaNPercentageFiltering(BothAxisFiltering):
+    """
+    Remove rows (default) or columns with a percentage of NaN values above a given percentage.
+    """
+
+    def __init__(self, dataframe: pd.DataFrame, percentage: Union[int, float] = 80, axis: int = 0):
+        """
+        :param percentage: maximum percentage of NaN values allowed (between 0 and 100)
+        :param axis: axis to apply filtering (index (0) or columns(1))
+        """
+        self.percentage_of_nan_allowed = percentage
+        super().__init__(dataframe, axis=axis)
+
+    def filter(self) -> pd.DataFrame:
+        thresh = self.df.shape[1-self.axis] * (self.percentage_of_nan_allowed/100)
```

Comment:
There is a confusion here I think for the `thresh`.

Definition of `thresh` from the documentation: *Require that many non-NA values.* -> in your current implementation you are doing the opposite since with give the percentage of how many NaN we are allowing. You need to either change the naming and documentation for `percentage` or change the way you compute the `thresh`.